Idea from the grub package in Mandrake:

don't add ``mem='' for 2.4 kernels :

Newer machines have several regions of reserved memory outside and
inside these regions.  grub's actions are suicide on newer laptops,
and machines like servers with lots of memory.  Laptops have special
sections of memory above 1MB which must be reserved... Ditto for
ACPI tables.  Using mem=XXX completely eliminates any information
that the BIOS has provided to the OS.

diff -ur grub-0.5.96.1.orig/stage2/builtins.c grub-0.5.96.1/stage2/builtins.c
--- grub-0.5.96.1.orig/stage2/builtins.c	Thu Oct 19 21:50:00 2000
+++ grub-0.5.96.1/stage2/builtins.c	Wed Mar 21 00:37:51 2001
@@ -2118,6 +2118,11 @@
   kernel_t suggested_type = KERNEL_TYPE_NONE;
   unsigned long load_flags = 0;
 
+  /*
+   * Don't pass a Linux's mem option automatically.
+   * If the kernel is another type, this flag has no effect.
+   */
+  load_flags |= KERNEL_LOAD_NO_MEM_OPTION;
   /* Deal with GNU-style long options.  */
   while (1)
     {
@@ -2147,9 +2152,12 @@
 	      return 1;
 	    }
 	}
-      /* If the `--no-mem-option' is specified, don't pass a Linux's mem
-	 option automatically. If the kernel is another type, this flag
-	 has no effect.  */
+      /* If the `--mem-option' is specified, pass a Linux's mem option.
+        If the `--no-mem-option' is specified, don't pass a Linux's mem
+	 option automatically. If the kernel is another type, this flags
+	 have no effect.  */
+      else if (grub_memcmp (arg, "--mem-option", 12) == 0)
+	load_flags &= ~KERNEL_LOAD_NO_MEM_OPTION;
       else if (grub_memcmp (arg, "--no-mem-option", 15) == 0)
 	load_flags |= KERNEL_LOAD_NO_MEM_OPTION;
       else
@@ -2184,14 +2192,15 @@
   "kernel",
   kernel_func,
   BUILTIN_CMDLINE,
-  "kernel [--no-mem-option] [--type=TYPE] FILE [ARG ...]",
+  "kernel [--mem-option] [--no-mem-option] [--type=TYPE] FILE [ARG ...]",
   "Attempt to load the primary boot image from FILE. The rest of the"
   "line is passed verbatim as the \"kernel command line\".  Any modules"
   " must be reloaded after using this command. The option --type is used"
   " to suggest what type of kernel to be loaded. TYPE must be either of"
   " \"netbsd\", \"freebsd\", \"openbsd\", \"linux\", \"biglinux\" and"
-  " \"multiboot\". The option --no-mem-option tells GRUB not to pass a"
-  " Linux's mem option automatically."
+  " \"multiboot\". The oprion --mem-option tells Grub to pass a Linux's"
+  " mem option automatically.  The option --no-mem-option tells GRUB"
+  " not to pass a Linux's mem option automatically (default)."
 };
 
 
